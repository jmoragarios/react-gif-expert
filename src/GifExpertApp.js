import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // const handleAdd = () => {

    //     const newCategory = 'Naruto';

    //     setcategories( [...categories, newCategory ]);

    // }

    return (
        <>
         <h2>Gif Expert App</h2>
         <AddCategory setCategories={ setCategories }/>
         <hr/>   

         <ol>
             {
                 categories.map( (category, i) => {
                     return <li key={ i }>{ category }</li>
                 })
             }
         </ol>
         
        </>
    )
}
