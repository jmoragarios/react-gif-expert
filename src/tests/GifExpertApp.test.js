import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp'

describe('Test <GifExpertApp/>', () => {
    
    test('should show show the component successfully', () => {
        
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();

    });
    

})
