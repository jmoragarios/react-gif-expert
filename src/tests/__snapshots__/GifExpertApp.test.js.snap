// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp/> should show show the component successfully 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
